addons/edi/models/edi_document.py

```diff
@@ -390,10 +423,36 @@ def commit_processing_state_change(self, processing_state=""):
         """
         self.ensure_one()
         if self.processing_state != processing_state:
+            self.update_processing_stats(processing_state)
             self.processing_state = processing_state
             self.env.cr.commit()
 
-    def create_or_update_edi_progress(self, log_message=None):
+    def update_processing_stats(self, processing_state):
+        """Track the start/end/total time for waiting/prepare/execute."""
+        EdiDocumentStats = self.env["edi.document.stats"]
+
+        self.ensure_one()
+        assert(self.processing_state != processing_state)
+        vals_stat = {}
+        now = fields.Datetime.now()
+        
+        # Stop the current state's processing time
+        current_state = self.processing_state or "waiting"
```

Comment:
Q: Did we want them to be able to have multiple lines of the same state for traceability? Maybe it's not needed..

Reply:
The DO for this one is `Start, end and total time of processing states: waiting, preparing, executing`, so I reckon having multiple lines for same state may not be required.